### ISSUE TITLE
As a CIF administrator, I want to see clear context of the changes in Amendments/General Revisions, so that I know what I am about to override

### ISSUE THREAD

**Issue by nanyangpro** [bcgov/cas-cif#1569]
#### Description:

As outcome of #1523 (with back-end work handled in #1528), the design of the diff section for an Amendment or General Revision is refined to provide clear context of the changes.

[- Design: Sequential Scenario 1](https://www.figma.com/file/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?node-id=7633%3A4258&t=9ZUtWsS7W7FVvhtv-1)
[- Design: Sequential Scenario 2](https://www.figma.com/file/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?node-id=7636%3A403&t=9ZUtWsS7W7FVvhtv-1)
[- Design: Sequential Scenario 3](https://www.figma.com/file/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?node-id=7636%3A669&t=9ZUtWsS7W7FVvhtv-1)

[- Design (Cleaner Version): preferred by users, nice-to-have, out of scope of this ticket, will be handled in #1573](https://www.figma.com/file/PzOY8RyQnVXnjZJdSv2AqQ/CIF---Internal-User-Portal?node-id=7647%3A912&t=9ZUtWsS7W7FVvhtv-1)

In summary, for the diff section, we'll show the following fields under their forms with help/contextual texts:
- The fields manually updated by the user
- The calculated and auto-populated fields impacted by the manual updates
- The fields changed by other
    - General Revision applied while the current Amendment is open (not approved)
    - Amendment approved while the current General Revision is open (not applied)

#### Acceptance Criteria:

Given I am a user
When I edit *non-calculated* fields on a form and another revision *hasn't* been committed since I opened the revision
Then when I view the "Forms updated" section
Then the old value is in black and struckout 
and if there was no old value (the field was blank) I see "Not entered"

Given I am a user
When I edit fields that affect *calculated* fields on a form and another revision *hasn't* been committed since I opened the revision
Then when I view the "Forms updated" section
Then the old value is in black and struckout 
and if there was no old value (the field was blank) I see "Not calculated"
and I see the help text (see wireframe)

Given I am a user
When I edit fields on a form and another revision *hasn't* been committed since I opened the revision
Then when I view the "Forms updated" section
Then the new value is in red
and there is an arrow between the old and new values

Given I am a user
When I edit fields on a form and another revision *has* been committed since I opened my revision, and the revisions *do not* touch the same fields
Then when I view the "Forms updated" section
I see the changes from my revision
and I see the changes from the other revision
and I see help text (see wireframe) below the changes from the other revision

Given I am a user
When I edit fields on a form and another revision *has* been committed since I opened my revision, and the revisions *do* touch the same fields
Then when I view the "Forms updated" section
I see the old value when I started my revision, then the other revision's value, then my new edited value (with colours and arrows in wireframe)
and I see help text (see wireframe)

NOTE: This ticket doesn't cover the yellow/green diffs. That page will be removed by https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1582

#### Development Checklist:

- [ ] Edit the work done in #1523 to match the design
- [ ] Add conditionality (#1523 shows the latest committed value for every field regardless of whether it changed in the revision)
- [ ] Tests (little testing was done in #1523 because we didn't have the ability to create more than one revision at the time)
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by Dianadec** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Climate-Action-Secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/1569?planning-poker) with Zenhub @BCerki @dleard @gurjmatharu @mikevespi @pbastia @sam-warren @Sepehr-Sobhani @tmastrom

**Comment by pbastia** [comment]
#1528 should have covered a lot of the pre-work for this card. Thanks @BCerki !

**Comment by gurjmatharu** [comment]
@nanyangpro is this issue meant for the `Review and Submit` page?  Wireframes show it being the `view` page.

**Comment by gurjmatharu** [comment]
|   old data	| new data	| committed data   	| example | result | 
|---	|---	|---	|---	|---	|
|   ✅	|   ❌	|   ❌	| Revision Type   	|    ?	| 
|   ✅	|   ✅	|   	|   Proposal Reference 	|    ~old data~ -> new data 	|  
|   ✅	|   ✅	|   ✅	|  2 edits happened	|  ?	|

**Comment by pbastia** [comment]
@gurjmatharu I think for the deleted data we've just used an italic _DELETED_ everywhere
For 2 edits happening in parallel, we have this in the prototype:
![image](https://user-images.githubusercontent.com/11507754/235198788-997b80a9-1ef1-47ee-aa28-da83bec6a4c4.png)

**Comment by BCerki** [comment]
Since this ticket isn't going to include the contextual help (?), we can probably roll that in with the tooltip implementation in https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1573

**Comment by nanyangpro** [comment]
The contextual help is the core of this ticket and should stay in the scope. #1573 is simply hiding them behind the tooltip icon.

**Comment by Dianadec** [comment]
Contextual text will be broken out into another card.

**Comment by gurjmatharu** [comment]
> Contextual text will be broken out into another card.

https://github.com/bcgov/cas-cif/issues/1739

**Comment by pbastia** [comment]
Thanks for the hard work @gurjmatharu ! This was a complex one.
One of the ACs can't be reproduced in dev:
> Given I am a user
When I edit fields on a form and another revision has been committed since I opened my revision, and the revisions do not touch the same fields
Then when I view the "Forms updated" section
I see the changes from my revision
and I see the changes from the other revision
and I see help text (see wireframe) below the changes from the other revision
